### PR TITLE
added main and launch files

### PR DESCRIPTION
**Pull request by archit2604**
added yaml file, launch file and main.py

### CODE REVIEW

**Comment by threewisemonkeys-as** [comment]
In `main.py` you will have to convert them into the correct formats. Ie.e robotstate etc..

**Comment by archit2604** [comment]
changed dataypes to gennav ones
also used lists in yaml file instead of tuples as yaml doesn't support them

**Comment by threewisemonkeys-as** [comment]
For planner and env also. You will need to actually instatiate the objects and then pass them to the commander. (You could use a dict for this)

**Comment by archit2604** [comment]
I'll do it for the planner but shouldn't the env one be the env_name and the parse_env will take care of it

**Comment by threewisemonkeys-as** [comment]
paser_env will take the env object. Otherwise you will have to pull the env parameters seperately to commander. So its just better to pass the whole object

**Comment by archit2604** [comment]
Ok will do

**Comment by archit2604** [comment]
Done

**Comment by threewisemonkeys-as** [comment]
what is `eval` here?

**Comment by archit2604** [comment]
It evaluates the expression which is passed to it as a string

**Comment by threewisemonkeys-as** [comment]
You could implement this also? 😅

Would be better to have a dict than a function

**Comment by archit2604** [comment]
I thought of that but the issue is every planner has a different set of parameters so even after taking them as a dict there would have to be a lot of if else cases for implementing that
I could do that but as the no of planners and env increase we'll have to  edit this code as well

**Comment by threewisemonkeys-as** [comment]
You can keep the params for the planner as fields in the yaml files right? Also to tackle the issue of having to rewrite, maybe you can write a function that converts the yaml file into a dict and passses that dict as `**kwargs` into the planner. This way, we will only have to keep adding any new params to the yaml file

**Comment by archit2604** [comment]
Yeah, this would work, so we would also have to change all planners to accept dicts as **kwargs and not the parameters itself, right?

**Comment by threewisemonkeys-as** [comment]
We dont need to change the planners at all. When you pass `**kwargs` to a function, it automatically matches the arguments.

https://www.programiz.com/python-programming/args-and-kwargs#:~:text=*args%20and%20*kwargs%20are%20special,to%20take%20variable%20length%20argument.&text=**kwargs%20passes%20variable%20number,a%20dictionary%20can%20be%20performed.

**Comment by archit2604** [comment]
Please Review the changes

**Comment by threewisemonkeys-as** [comment]
Can you format with `black` once?
Also do the complete the imports instead of using `*`

Looks good apart from that

**Comment by archit2604** [comment]
Done

File: src/main.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+env_name = rospy.get_param("env_name")
+obstacles = rospy.get_param("obstacle_data")
+for planner_chk in planner_dict:
+    if planner_name == planner_chk:
+        planner = planner_dict[planner_name]
+        break
+for env_chk in env_dict:
+    if env_name == env_chk:
+        env = env_dict[env_name]
+        break
```
Comment (threewisemonkeys-as) [inline src/main.py]: You can directly check if an object is one of the keys of a dict using `key in dict.keys()` i think. No need to loop over all the keys

**Review by threewisemonkeys-as — DISMISSED** [review]

File: src/main.py (hunk 1 above)
Comment (archit2604) [reply src/main.py]: Ya but this also assigns the value of the key to planner which is the used in the next set of lines

Comment (threewisemonkeys-as) [reply src/main.py]:
```python
if planner_name in planner_dict.keys():
    planner = planner_dict[planner_name]
```

I mean this. Would to the same job right?

Comment (archit2604) [reply src/main.py]: Ohh yes,my bad

Comment (archit2604) [reply src/main.py]: I'll change it and commit tomorrow

**Review by threewisemonkeys-as — APPROVED** [review]

**Pull request merged**